owltorrent-storage/src/main/java/edu/rice/owltorrent/storage/DownloadFile.java

```diff
@@ -0,0 +1,98 @@
+package edu.rice.owltorrent.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+
+/**
+ * Represents the local disk version of a file that a client is downloading from the BitTorrent
+ * network.
+ */
+public class DownloadFile {
+
+  static class FileAlreadyExistsException extends Exception {}
+
+  static class FileCouldNotBeCreatedException extends Exception {}
+
+  static class IllegalByteOffsets extends Exception {}
```

Comment:
In the Common package we definitely want a subpackage for exceptions. So 2 choices:
1. Write all of them there. (Not a must since these have no content? But we might want to have a class like DownloadException and have these extend it, or just have them all be DownloadExceptions with different error codes. )
2. Use public modifier instead.
Feel free to take the call on this one. 

**But we definitely want to extend RunTimeException instead.**

Reply:
Decided to make them public and keep them in the same class.